Python/Product/VSCode/AnalysisVsc/PathsWatcher.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.DsTools.Core.Disposables;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+
+namespace Microsoft.Python.LanguageServer {
+    internal sealed class PathsWatcher : IDisposable {
```

Comment:
I thought  that VSC is a source of truth in terms of files.

Reply:
TS file watcher is much less reliable as I found. Also, this is generally useful behavior. If we are to do it at TS level, we'd need custom `python/reloadModules` notification.

Reply:
Ok, so how do we resolve conflicts with `workspace/didChangeWatchedFiles`?

Reply:
It never come AFAICT b/c we don't have any watched files registered through LSP

Reply:
Is it possible that we end up with set of files and folders that is different from what user sees in VSC?

Reply:
Yes. Python folders are different from user code. I.e. we are watching Python search (i.e libraries) while VSC sees user code. In RTVS we watched `install.packages` while here we watch for folder changes.

Reply:
Ok, so we don't actually care about client side but rather interesting in real state of file system.